lib/fun_ruby/modules/enum/max.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module FunRuby
+  module Modules
+    # @private
+    module Enum
+      # Returns the minimum element of the enumerable
+      #
+      # @since 0.1.0
+      #
+      # @param enumerable [#to_enum]
+      #
+      # @return [Object]
+      #
+      # @example Basics
+      #   F::Modules::Enum.max([1, 2, 3]) # => 3
+      #   F::Modules::Enum.max([]) # => nil
+      #
+      # @example Curried
+      #   curried = F::Modules::Enum.max
+      #   curried.([1, 2, 3]) # => 3
+      #   curried.([]) # => nil
+      def max(enumerable = F._)
```

Comment:
Let's adjust the methods `min` and `max` to accept a function that is going to decide which element is max by returning `-1`, `0` and `1`.

So, it will look like this:

```ruby
collection_of_numbers = [1, 2, 3]
collection_of_strings = ['aaa', 'aa', 'a']

cmp_string = -> (first, second) { first.size <=> second.size }
cmp_number = ->(first, second) { first <=> second }

longest_string = F::Modules::Enum.max(cmp_string)
biggest_number = F::Modules::Emum.max(cmd_number)

longest_string.(collection_of_strings) # => 'aaa'
biggest_number.(collection_of_numbers) # => 3
```

Reply:
Okay, I see.